solarwinds_apm/inbound_metrics_processor.py

```diff
@@ -141,12 +170,25 @@ def calculate_transaction_names(
         url_tran = span.attributes.get(self._HTTP_URL, None)
         http_route = span.attributes.get(self._HTTP_ROUTE, None)
         trans_name = None
-        if http_route:
+        custom_trans_name = self.calculate_custom_transaction_name(span)
+
+        if custom_trans_name:
+            trans_name = custom_trans_name
+        elif http_route:
             trans_name = http_route
         elif span.name:
             trans_name = span.name
         return trans_name, url_tran
 
+    def calculate_custom_transaction_name(self, span: "ReadableSpan") -> Any:
+        """Get custom transaction name for trace by trace_id, if any"""
+        trans_name = None
+        trace_span_id = f"{span.context.trace_id}-{span.context.span_id}"
+        custom_name = self.apm_txname_manager.get(trace_span_id)
+        if custom_name:
+            trans_name = self.apm_txname_manager[trace_span_id]
```

Comment:
i'm not sure since not too familiar with python but couldn't this be
```suggestion
            trans_name = custom_name
```

Reply:
🙃 Yup! Addressed in [160180d](https://github.com/solarwindscloud/solarwinds-apm-python/pull/115/commits/160180d5ba6da0cf8638a6a642515864553ce23d) 😺